alchemy/src/cloudflare/r2-rest-state-store.ts

```diff
@@ -70,6 +67,45 @@ export class R2RestStateStore implements StateStore {
     // Create Cloudflare API client with automatic account discovery
     this.api = await createCloudflareApi(this.options);
 
+    // Check if the bucket exists
+    const response = await withExponentialBackoff(
+      async () => {
+        const response = await this.api.get(
+          `/accounts/${this.api.accountId}/r2/buckets/${this.bucketName}`
+        );
+
+        if (!response.ok) {
+          await handleApiError(response, "get", "bucket", this.bucketName);
+        }
+
+        return response;
+      },
+      isRetryableError,
+      5,
+      1000
```

Comment:
Nit: extract this out into `getBucket` in `bucket.ts`

Trying to build up raw fetch handlers as we go.

Reply:
Heck yeah man!  Working on a test now...

Reply:
Resolved via f500b60c0fe1a4221da86452add26fe8e5fd8f52